src/Result/Invoice.php

```diff
@@ -24,6 +24,51 @@ class Invoice extends AbstractResult
 
     public const ADDITIONAL_STATUS_PAID_LATE = 'PaidLate';
 
+    public function getId(): string
+    {
+        return $this->getData()['amount'];
+    }
+
+    public function getAmount(): float
+    {
+        return $this->getData()['amount'];
```

Comment:
Not sure if we should keep it as string as with PreciseNumber to avoid any big number issues. Your thoughts/experience @woutersamaey ?

Reply:
@woutersamaey @utxo-one The reason I mention it is because I recently needed[ BC Math functions](https://www.php.net/manual/en/ref.bc.php) and they all use strings for the return type to ensure it stays consistent. E.g. if you have BTC as currency and the invoice amount is `0.00002100` BTC it would return `2.1E-5` which is still fine and you can print it as string again but it is kinda weird maybe. Or maybe even better, we stick with `PreciseNumber` like we do [elsewhere](https://github.com/btcpayserver/btcpayserver-greenfield-php/blob/master/src/Result/PullPaymentPayout.php#L59)?

Reply:
> Not sure if we should keep it as string as with PreciseNumber to avoid any big number issues. Your thoughts/experience @woutersamaey ?

I prefer the PreciseNumber as this was meant to be complementary to bcmath as it keeps the data inside as a string.

Reply:
> @woutersamaey @utxo-one The reason I mention it is because I recently needed[ BC Math functions](https://www.php.net/manual/en/ref.bc.php) and they all use strings for the return type to ensure it stays consistent. E.g. if you have BTC as currency and the invoice amount is `0.00002100` BTC it would return `2.1E-5` which is still fine and you can print it as string again but it is kinda weird maybe. Or maybe even better, we stick with `PreciseNumber` like we do [elsewhere](https://github.com/btcpayserver/btcpayserver-greenfield-php/blob/master/src/Result/PullPaymentPayout.php#L59)?

Use PreciseNumber yep!

Reply:
Ok great I changed it. Hopefully I did it right!

Reply:
Almost perfect, you need to return it like this https://github.com/btcpayserver/btcpayserver-greenfield-php/blob/master/src/Result/PullPaymentPayout.php#L62

When this is fixed, ready to merge 👍

Reply:
The PreciseNumber class may not look like much, but if you remember @ndeet we had some `Math` class that was a wrapper for `bcmath` and that used these classes exclusively. I think we dropped it at some point as it was not really used, but this does provide some background and direction for the `PreciseNumber` class.

Reply:
Yes remembered after looking into this and why float sounded some internal alarm when I saw it :D, good to have agreement on this. 👍

Reply:
Updated to return parseString()